src/lesson/lesson01.js

```diff
@@ -262,6 +262,170 @@ Lesson01ExerciseStepPlayer.prototype = {
+          grid_cell_size, stepPlayer.game.n_cols);
+      grid.x = offset_x;
+      grid.y = offset_y;
+      this._animator.addElement(grid);
+
+      var character = new ElementFactories.createRobot(
+          'p' + sp, grid_cell_size, grid_cell_size);
+      character.x = offset_x + (0.5 + stepPlayer.game.character_position) *
+          grid_cell_size;
+      character.y = offset_y + (0.5 + stepPlayer.game.n_rows - 1) *
+          grid_cell_size;
+      this._animator.addElement(character);
+
+      for (var i = 1; i < stepPlayer.game.n_rows; ++i) {
+        for (var obstacle_pos in stepPlayer.game.obstacles[i]) {
+          var obstacle = new Animator.CircleElement(
+              'o' + sp + i + obstacle_pos, grid_cell_size / 2);
+          obstacle.x = offset_x +
+              (0.5 + stepPlayer.game.obstacles[i][obstacle_pos]) *
+              grid_cell_size;
+          obstacle.y = offset_y + (0.5 + (stepPlayer.game.n_rows - 1 - i)) *
+              grid_cell_size;
+          this._animator.addElement(obstacle);
+        }
+      }
+    }
+  },
+  _render: function(actions) {
+    console.log("Entered here.")
+    nPlayers = this.lessonStepPlayers.length;
+    var directions_failure_reason = this._actionsToDirections(actions);
+    for (var sp in this.lessonStepPlayers) {
+      var stepPlayer = this.lessonStepPlayers[sp];
+      var directions = directions_failure_reason[0];
+      var failure_reason = directions_failure_reason[1];
+      var grid_cell_size = stepPlayer.game.gridCellSize() / nPlayers;
+      var failure_collector = Array();
+
+      for (var i = 1; i < stepPlayer.game.obstacles.length; ++i) {
+        for (var obstacle_pos in stepPlayer.game.obstacles[i]) {
+          this._animator.addAnimation(AnimationFactories.straightMove(
+                'o' + sp + i + obstacle_pos, 0, directions.length,
+                0, grid_cell_size * directions.length));
+
+          var offset_y = (Constants.RUN_VIEW_SQUARE_DIMENSION / 2 -
+                          grid_cell_size * stepPlayer.game.n_rows / 2);
+
+          this._animator.addAnimation(
+              new Animator.Animation(0, directions.length,
+                'o' + sp + i + obstacle_pos, 'radius',
+                function() {
+                  var max_y = offset_y + this.game.n_rows * grid_cell_size;
+                  function o_radius_fn(t, elem) {
+                    return elem.y <= max_y ? elem.radius : 0;
+                  }
+                  return o_radius_fn;
+                }.bind(stepPlayer)()));
+        }
+      }
+
+      var character = this._animator.getElement('p' + sp);
+
+      for (var i = 0; i < directions.length; i++) {
+        var animationName = ((directions[i] == -1) ? 'walk_left':
+                             (directions[i] == 1) ? 'walk_right' : 'walk_down');
+        var duration = (animationName == 'walk_down' ) ? 0.1 : 1;
+        console.log("Directions[" + i + "] = " + directions[i]);
```

Comment:
The other code left this log message behind. You can remove both (this one and the one in the other _render method).